static/app/components/tours/components.tsx

```diff
@@ -0,0 +1,262 @@
+import {IconClose} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {useHotkeys} from 'sentry/utils/useHotkeys';
+import useOverlay, {type UseOverlayProps} from 'sentry/utils/useOverlay';
+
+export interface TourContextProviderProps<T extends TourEnumType> {
+  /**
+   * The children of the tour context provider.
+   * All children of this component will be blurred when the tour is active.
+   * Only the active tour element and overlay will be discernible.
+   */
+  children: React.ReactNode;
+  /**
+   * Whether the tour can be accessed by the user
+   */
+  isAvailable: TourState<T>['isAvailable'];
+  /**
+   * The ordered list of Step IDs
+   */
+  orderedStepIds: TourState<T>['orderedStepIds'];
+  /**
+   * The React context (from createContext) containing the provider for the tour.
+   * The value for this prop comes from useTourReducer, to avoid extra steps.
+   */
+  tourContext: React.Context<TourContextType<T>>;
+  /**
+   * Whether to omit the blurring window.
+   */
+  omitBlur?: boolean;
+}
+
+export function TourContextProvider<T extends TourEnumType>({
+  children,
+  isAvailable,
+  tourContext,
+  omitBlur,
+  orderedStepIds,
+}: TourContextProviderProps<T>) {
+  const tourContextValue = useTourReducer<T>({
+    isAvailable,
+    isRegistered: false,
+    orderedStepIds,
+    currentStepId: null,
+  });
+  const {dispatch, currentStepId} = tourContextValue;
+  const isTourActive = currentStepId !== null;
+
+  const tourHotkeys = useMemo(() => {
+    return [
+      {match: 'Escape', callback: () => dispatch({type: 'END_TOUR'})},
+      {match: ['left', 'h'], callback: () => dispatch({type: 'PREVIOUS_STEP'})},
+      {match: ['right', 'l'], callback: () => dispatch({type: 'NEXT_STEP'})},
+    ];
+  }, [dispatch]);
+
+  useHotkeys(tourHotkeys);
+
+  return (
+    <tourContext.Provider value={tourContextValue}>
+      {isTourActive && !omitBlur && <BlurWindow data-test-id="tour-blur-window" />}
+      {children}
+    </tourContext.Provider>
+  );
+}
+
+interface BaseTourElementProps<T extends TourEnumType> extends Partial<UseOverlayProps> {
+  /**
+   * The content being focused during the tour.
+   */
+  children: React.ReactNode;
+  /**
+   * The description of the tour step.
+   */
+  description: React.ReactNode;
+  /**
+   * The unique identifier of the tour step.
+   */
+  id: TourStep<T>['id'];
+  /**
+   * The title of the tour step.
+   */
+  title: React.ReactNode;
+  /**
+   * The relevant tour context
+   */
+  tourContext: TourContextType<T>;
+  /**
+   * The className of the wrapper element.
+   */
+  className?: string;
+}
+
+/**
+ * Most implementations of TourElementProps will hook in their own context.
+ */
+export type TourElementProps<T extends TourEnumType> = Omit<
+  BaseTourElementProps<T>,
+  'tourContext'
+>;
+
+export function TourElement<T extends TourEnumType>({
+  children,
+  id,
+  title,
+  description,
+  tourContext,
```

Comment:
Could this be accessed internally by doing `useTourContext(id)`?

This way someone could just directly use TourElement without having to create an extra wrapper that just accesses this context (similar to the one you used in the test file)

Reply:
Refactored to pass in the context like you mentioned!